drivers/SmartThings/zigbee-smoke-detector/src/test/test_frient_smoke_detector.lua

```diff
@@ -15,182 +15,460 @@
+                        0x00
+                )
+            })
+
+            test.socket.zigbee:__expect_send({
+                mock_device.id,
+                IASWD.attributes.MaxDuration:write(mock_device, ALARM_DEFAULT_MAX_DURATION)
+            })
+
+            mock_device:expect_metadata_update({ provisioning_state = "PROVISIONED" })
+        end
+)
+
+test.register_message_test(
+        "Battery voltage report should be handled",
+        {
+            {
+                channel = "zigbee",
+                direction = "receive",
+                message = { mock_device.id, PowerConfiguration.attributes.BatteryVoltage:build_test_attr_report(mock_device, 24) }
+            },
+            {
+                channel = "capability",
+                direction = "send",
+                message = mock_device:generate_test_message("main", capabilities.battery.battery(14))
+            }
+        }
+)
+
+test.register_message_test(
+        "ZoneStatusChangeNotification should be handled: detected",
+        {
+            {
+                channel = "zigbee",
+                direction = "receive",
+                -- ZoneStatus | Bit0 Alarm1 set to 1
+                message = { mock_device.id, IASZone.client.commands.ZoneStatusChangeNotification.build_test_rx(mock_device, 0x0001, 0x00) }
+            },
+            {
+                channel = "capability",
+                direction = "send",
+                message = mock_device:generate_test_message("main", capabilities.smokeDetector.smoke.detected())
+            }
+        }
+)
+
+test.register_message_test(
+        "ZoneStatusChangeNotification should be handled: tested",
+        {
+            {
+                channel = "zigbee",
+                direction = "receive",
+                -- ZoneStatus | Bit8: Test set to 1
+                message = { mock_device.id, IASZone.client.commands.ZoneStatusChangeNotification.build_test_rx(mock_device, 0x100, 0x01) }
+            },
+            {
+                channel = "capability",
+                direction = "send",
+                message = mock_device:generate_test_message("main", capabilities.smokeDetector.smoke.tested())
+            }
+        }
+)
+
+test.register_message_test(
+        "Temperature report should be handled (C) for the temperature cluster",
+        {
+            {
+                channel = "zigbee",
+                direction = "receive",
+                message = { mock_device.id, TemperatureMeasurement.attributes.MeasuredValue:build_test_attr_report(mock_device, 2500) }
+            },
+            {
+                channel = "capability",
+                direction = "send",
+                message = mock_device:generate_test_message("main", capabilities.temperatureMeasurement.temperature({ value = 25.0, unit = "C" }))
+            }
+        }
 )
 
 test.register_coroutine_test(
-    "Configure should configure all necessary attributes",
```

Comment:
I'd like to see an updated version of this test that reflects the new expected results